### PR TITLE
another random maint expansion: making maint a tad more dangerous

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
- adds 34 new random maint rooms, credit to GizkaFreechman for most. 
- new gimmick job, the VIP. the VIP is rich.
- adds random maint traps: invisible traps that may or may not spawn. they range from breaking all the lights in a room to spawning mobs nearby
- also removes knockdown from faithless while i was in his files cuz i forgot it awhile ago

## Why It's Good For The Game
I hate you all, and i want to remind you that no one is fucking safe

## Changelog
:cl: Zeskorion, GizkaFreechman, thatguythere03
add: 28 random maint rooms by GizkaFreechman, edited by myself
add: Random maint traps
add: cluwne rune, which can be used by cultists in a manner similar to an invocation rune, if the Honkmother has blessed one of the invokers with her presence
add: new random maint job, the VIP. he's rich, apparently important, and actually probably quite useless.
imageadd: sprites for new job and cluwne rune by thatguythere03
tweak: edited most random maint maps to make maint easier to get through (changing table placements and such)
balance: faithless no longer has a 12% chance to stun for 6 seconds on hit
/:cl:

### CODE REVIEW

File: code/game/objects/effects/mainttraps.dm [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+						qdel(src)
+
+/obj/effect/trap/trigger/TrapEffect(AM)
+	if(inuse)
+		return FALSE
+	else 
+		inuse = TRUE
+	var/list/possibletraps = list()
+	for(var/obj/effect/trap/nexus/payload in oview(10, src))
+		possibletraps += payload
```
Comment (ike709) [inline code/game/objects/effects/mainttraps.dm]: This list should probably just be populated once in LateInitialize rather than every time. `oview()` is expensive.

File: code/game/objects/effects/mainttraps.dm [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+	inuse = FALSE
+	return TRUE
+
+/obj/effect/trap/nexus/cluwnecurse
+	name = "honk :0)"
+	desc = "oh god..."
+
+/obj/effect/trap/nexus/cluwnecurse/TrapEffect(AM)
+	var/turf/T = get_turf(src)
+	for(var/mob/living/carbon/human/C in oview(5, src))
```
Comment (ike709) [inline code/game/objects/effects/mainttraps.dm:130]: might be better to use can_see()

**Comment by ike709** [comment]
Christ I had to stop looking at the code. You use `view()` and `oview()` *way* too much.

**Comment by ike709** [comment]
You should look at `fields.dm` and consider using fields for this. They shouldn't be too expensive since they shouldn't really be moving or processing.

**Comment by zeskorion** [comment]
Aight, ill see about that when i get home them

**Comment by zeskorion** [comment]
Dammit gizka stop using step vars

**Comment by imw-Oculus** [comment]
> also removes knockdown from faithless while i was in his files cuz i forgot it awhile ago
> new gimmick job, the VIP. the VIP is rich.

What do these have to do with Maint? Atomize your PR's. This is the third PR of yours in a row where you've just blatantly threw in completely unrelated things into a PR. Your PR's aren't going to get merged if they're not atomized.

> adds 34 new random maint rooms, credit to GizkaFreechman for most.

This is great! In concept, at least - haven't reviewed the particular rooms yet.

> adds random maint traps: invisible traps that may or may not spawn. they range from breaking all the lights in a room to spawning mobs nearby

I like the spirit of adding traps - but I don't think they should be invisible. Give the player an opportunity to avoid them, otherwise it's just "Whoops, looks like you made the mistake of trying to explore maint! Time to die.". Good systems give the player an opportunity to interact with them.

**Comment by zeskorion** [comment]
> > also removes knockdown from faithless while i was in his files cuz i forgot it awhile ago
> > new gimmick job, the VIP. the VIP is rich.
> 
> What do these have to do with Maint? Atomize your PR's. This is the third PR of yours in a row where you've just blatantly threw in completely unrelated things into a PR. Your PR's aren't going to get merged if they're not atomized.

nah i saw it while doing spawnercode, i'd missed it in a previous pr. Gimmick jobs, however, are a random maint thing. every random maint pr since they were introduced has had one

> > adds random maint traps: invisible traps that may or may not spawn. they range from breaking all the lights in a room to spawning mobs nearby
> 
> I like the spirit of adding traps - but I don't think they should be invisible. Give the player an opportunity to avoid them, otherwise it's just "Whoops, looks like you made the mistake of trying to explore maint! Time to die.". Good systems give the player an opportunity to interact with them.

none of the traps will directly kill the players. There are, at the moment, four kinds of trap: lightbreaker, which breaks lights, doorbolter, which bolts doors for about ten seconds, spawners, which try to spawn a mob out of sight after being triggered a certain amount of times, and cluwne curse, which is quite rare, and spawns a nonlethal floorcluwne

**Comment by utilbird** [comment]
>non lethal floor clown

Lame

**Comment by zeskorion** [comment]
> > non lethal floor clown
> 
> Lame

you can make it lethal through a very difficult and convoluted ritual

**Comment by TeomanTheGreat** [comment]
> 
> 
> > > non lethal floor clown
> > 
> > 
> > Lame
> 
> you can make it lethal through a very difficult and convoluted ritual

convoluted ritual you say? explain

**Comment by zeskorion** [comment]
> > > > non lethal floor clown
> > > 
> > > 
> > > Lame
> > 
> > 
> > you can make it lethal through a very difficult and convoluted ritual
> 
> convoluted ritual you say? explain

first, you need to find one of the incredibly rare cluwne runes. ill let you figure out the rest

**Comment by utilbird** [comment]
Could give VIP some very expensive item that cargo will probably want, with some mood debuff and/or crew objective red text if they lose it

**Comment by zeskorion** [comment]
> Could give VIP some very expensive item that cargo will probably want, with some mood debuff and/or crew objective red text if they lose it

Their pda is worth 500, and both envirosuit and helmet for vip plasmaman are worth 4500 each

File: code/modules/mob/living/simple_animal/hostile/faithless.dm [hunk 3]
```diff
@@ -34,11 +34,3 @@
...
 	do_footstep = TRUE
 	hardattacks = TRUE
-
-/mob/living/simple_animal/hostile/faithless/AttackingTarget()
-	. = ..()
-	if(. && prob(12) && iscarbon(target))
-		var/mob/living/carbon/C = target
-		C.Paralyze(60)
-		C.visible_message("<span class='danger'>\The [src] knocks down \the [C]!</span>", \
-				"<span class='userdanger'>\The [src] knocks you down!</span>")
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/hostile/faithless.dm:44]: Why is this removed?

Comment (zeskorion) [reply code/modules/mob/living/simple_animal/hostile/faithless.dm:44]: I was in the code for it and missed it in a previous pr so i decided to hit it

**Comment by PowerfulBacon** [comment]
Is this ready for a testmerge?

**Comment by zeskorion** [comment]
> Is this ready for a testmerge?

Yes

File: code/game/objects/effects/mainttraps.dm [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+				if(TrapEffect(AM))
+					if(!reusable)
+						qdel(src)
+
+/obj/effect/trap/trigger/TrapEffect(AM)
+	if(inuse)
+		return FALSE
+	else 
+		inuse = TRUE
+	if(!possibletraps.len)
```
Comment (PowerfulBacon) [inline code/game/objects/effects/mainttraps.dm]: !LAZYLEN(possibletraps)

File: code/game/objects/effects/mainttraps.dm [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+/obj/effect/trap/proc/TrapEffect(AM)
+	return TRUE
+
+/obj/effect/trap/trigger //this only triggers other traps
+	name = "pressure pad"
+	desc = "seems flimsy"
+	var/grounded = FALSE //does it ignore fliers
+	var/pick_style = PICK_STYLE_ORDERED
+	var/requirehuman = TRUE
+	var/list/possibletraps = list()
```
Comment (PowerfulBacon) [inline code/game/objects/effects/mainttraps.dm:23]: Apparently it's faster to initialize lists in the Initialize proc of objects than in the definition

**Comment by PowerfulBacon** [comment]
Testmerged onto sage

**Comment by qwertyquerty** [comment]
Fix conflicts

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
Too many maps so mapdiffs isn't working, but from watching it all of the afters generated fine and it worked when testmerged, so going to assume this to be ok
